spring-core/src/test/java/org/springframework/core/retry/support/MaxRetryAttemptsPolicyTests.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2002-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.core.retry.support;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import org.springframework.core.retry.RetryExecution;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * Tests for {@link MaxRetryAttemptsPolicy}.
+ *
+ * @author Mahmoud Ben Hassine
+ */
+class MaxRetryAttemptsPolicyTests {
+
+	@Test
+	void testDefaultMaxRetryAttempts() {
+		// given
+		MaxRetryAttemptsPolicy retryPolicy = new MaxRetryAttemptsPolicy();
+
+		// when
+		RetryExecution retryExecution = retryPolicy.start();
+
+		// then
+		assertThat(retryExecution.shouldRetry(Mockito.any(Exception.class))).isTrue();
```

Comment:
You still use `Mockito.any()` in these tests while it is wrong context for this API.
Here the simple `Mockito.mock()` should be enough.
Or might be even better from performance perspective to use some real exception instance instead of reflection 😄

Reply:
You are right, apologies I just missed that from our last review. Should be fixed now.